test/unit_tests/logging/CMakeLists.txt

```diff
@@ -0,0 +1,143 @@
+  list(APPEND unit_tests "test_logging_arkode_arkstep.cpp\;1 1 0") # DIRK Newton
+                                                                   # + GMRES
+  list(APPEND unit_tests "test_logging_arkode_arkstep.cpp\;1 0") # DIRK
+                                                                 # Fixed-point
+  list(APPEND unit_tests "test_logging_arkode_arkstep.cpp\;2 1 1") # ImEx Newton
+                                                                   # + Dense
+  list(APPEND unit_tests "test_logging_arkode_arkstep.cpp\;2 1 0") # ImEx Newton
+                                                                   # + GMRES
+  list(APPEND unit_tests "test_logging_arkode_arkstep.cpp\;2 0") # ImEx
+                                                                 # Fixed-point
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;0") # Ex-MRI
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;1 1 1") # Im-MRI
+                                                                   # Newton +
+                                                                   # Dense
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;1 1 0") # Im-MRI
+                                                                   # Newton +
+                                                                   # GMRES
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;1 0") # Im-MRI
+                                                                 # Fixed-point
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;2 1 1") # ImEx-MRI
+                                                                   # Newton +
+                                                                   # Dense
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;2 1 0") # ImEx-MRI
+                                                                   # Newton +
+                                                                   # GMRES
+  list(APPEND unit_tests "test_logging_arkode_mristep.cpp\;2 0") # ImEx-MRI
+                                                                 # Fixed-point
+endif()
+
+if(BUILD_CVODE)
+  list(APPEND unit_tests "test_logging_cvode.cpp\;1 1") # Newton + Dense
+  list(APPEND unit_tests "test_logging_cvode.cpp\;1 0") # Newton + GMRES
+  list(APPEND unit_tests "test_logging_cvode.cpp\;0") # Fixed-point
+endif()
+
+if(BUILD_CVODES)
+  list(APPEND unit_tests "test_logging_cvodes.cpp\;1 1") # Newton + Dense
+  list(APPEND unit_tests "test_logging_cvodes.cpp\;1 0") # Newton + GMRES
+  list(APPEND unit_tests "test_logging_cvodes.cpp\;0") # Fixed-point
+endif()
+
+if(BUILD_IDA)
+  list(APPEND unit_tests "test_logging_ida.cpp\;1") # Newton + Dense
+  list(APPEND unit_tests "test_logging_ida.cpp\;0") # Newton + GMRES
+endif()
+
+if(BUILD_IDAS)
+  list(APPEND unit_tests "test_logging_idas.cpp\;1") # Newton + Dense
+  list(APPEND unit_tests "test_logging_idas.cpp\;0") # Newton + GMRES
+endif()
```

Comment:
This is a *ton* of new tests; multiple of which seem redundant.  I'm unclear on whether the CI or Jenkins will run all examples/tests for all logging levels (and all precisions), but that seems to be even more redundant.

Would it be possible to get full "coverage" with a smaller set of tests so that we don't overwhelm the CI bots or Jenkins?  Could these just be run at the highest logging level (since that output includes all of the lower logging levels)?

Reply:
Unfortunately, this is as lean as I think the set of tests can be. The different logging levels need to be tested separately to ensure output from a higher level is not accidentally output at a lower level e.g., debugging information printed when only logging errors. The GitHub logging test (`.github/workflows/ubuntu-clang-latest.yml`) was updated to build with each logging level and run only these logging tests (this PR also adds a `LABELS` option to `sundials_add_test` so we can run only the tests with a given label) to keep the testing time down. Testing each logging level takes about 3 minutes and can be run in parallel to each other. The other CI tests are unchanged and only run with the default logging level. The different combinations of intergrator/stepper and linear/nonlinear solver used I believe are the minimal set needed to cover the code different pathways and logging output nesting levels.

Reply:
That sounds good; I was concerned that the total number of tests might multiply by a factor of 5, but with these labeled appropriately it isn't onerous.